src/order/sup_indep.lean

```diff
@@ -14,6 +14,11 @@ sup-independent if, for all `a`, `f a` and the supremum of the rest are disjoint
 
 In distributive lattices, this is equivalent to being pairwise disjoint.
 
+## Implementation notes
+
+We avoid the "obvious" definition `∀ i ∈ s, disjoint (f i) ((s.erase i).sup f)` because `erase`
```

Comment:
I think this comment is too far away from the definition. Please repeat it in the docstring.